common/persistence/client/fx.go

```diff
@@ -201,6 +205,10 @@ func DataStoreFactoryProvider(
 		dataStoreFactory = faultinjection.NewFaultInjectionDatastoreFactory(defaultStoreCfg.FaultInjection, dataStoreFactory)
 	}
 
+	if otel.IsEnabled(tracerProvider) {
+		dataStoreFactory = telemetry.NewTelemetryDataStoreFactory(dataStoreFactory, tracerProvider.Tracer("persistence"))
+	}
```

Comment:
No OTEL, no interceptor. That's why the generated tracing code doesn't check again if it's recording or not.